### PR TITLE
Tests and code improvement in avs registry writer

### PR DESCRIPTION
**Pull request by pablodeymo**
- Tests in `avs_registry` `writer`
- Refactor and code improvment in `avs_registry`.

### CODE REVIEW

File: crates/chainio/clients/avsregistry/src/writer.rs [hunk 1]
```diff
@@ -284,13 +317,141 @@ impl AvsRegistryChainWriter {
...
+    /// Retrieves the status of a transaction from its hash.
+    ///
+    /// # Arguments
+    ///
+    /// `tx_hash` - The hash of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// A bool indicating wether the transaction was successful or not.
+    async fn get_transaction_status(tx_hash: FixedBytes<32>) -> bool {
```
Comment (supernovahs) [inline crates/chainio/clients/avsregistry/src/writer.rs:369]: should we move this to testing utils? https://github.com/Layr-Labs/eigensdk-rs/blob/main/testing/testing-utils/src/anvil_constants.rs

File: crates/chainio/clients/avsregistry/src/writer.rs [hunk 2]
```diff
@@ -284,13 +317,141 @@ impl AvsRegistryChainWriter {
...
+    use alloy_provider::Provider;
+    use eigen_crypto_bls::BlsKeyPair;
+    use eigen_logging::get_test_logger;
+    use eigen_testing_utils::anvil_constants::{
+        get_operator_state_retriever_address, get_registry_coordinator_address, ANVIL_RPC_URL,
+    };
+    use std::time::Duration;
+    use tokio::time::sleep;
+
+    const ANVIL_HTTP_URL: &str = "http://localhost:8545";
```
Comment (supernovahs) [inline crates/chainio/clients/avsregistry/src/writer.rs:343]: https://github.com/Layr-Labs/eigensdk-rs/blob/2d455ccb25743f5f79aef56a20e85adf0e330f95/testing/testing-utils/src/anvil_constants.rs#L20

File: crates/chainio/clients/avsregistry/Cargo.toml [hunk 3]
```diff
@@ -27,10 +27,13 @@ alloy-network.workspace = true
...
 
 [dev-dependencies]
-tokio = {version = "1.37.0" , features = ["test-util", "full","sync"] }
+tokio = { version = "1.37.0", features = ["test-util", "full", "sync"] }
 hex = "0.4.3"
-once_cell.workspace = true
\ No newline at end of file
+once_cell.workspace = true
+eigen-testing-utils.workspace = true
+alloy-node-bindings.workspace = true
```
Comment (supernovahs) [inline crates/chainio/clients/avsregistry/Cargo.toml:39]: alphabetical nit

**Review by supernovahs — CHANGES_REQUESTED** [review]

File: crates/chainio/clients/avsregistry/Cargo.toml (hunk 3 above)
Comment (supernovahs) [reply crates/chainio/clients/avsregistry/Cargo.toml:39]: PS : https://crates.io/crates/cargo-abc :)

Comment (pablodeymo) [reply crates/chainio/clients/avsregistry/Cargo.toml]: done 👍

File: crates/chainio/clients/avsregistry/src/writer.rs (hunk 2 above)
Comment (pablodeymo) [reply crates/chainio/clients/avsregistry/src/writer.rs]: done! 
also generalized all this constants of the whole project in the file anvil_constants.rs

File: crates/chainio/clients/avsregistry/src/writer.rs (hunk 1 above)
Comment (pablodeymo) [reply crates/chainio/clients/avsregistry/src/writer.rs]: done!

**Pull request merged**